funsor/sum_product.py

```diff
@@ -267,7 +266,7 @@ def shift_funsor(f, t):
     if not lags:
         return naive_sequential_sum_product(sum_op, prod_op, trans, time_var, {})
 
-    period = int(np.lcm.reduce(list(lags)))
+    period = int(reduce(lambda a, b: a * b // gcd(a, b), list(lags)))
```

Comment:
`np.lcm.reduce` is not available in the latest version of Numpy that supports Python 3.5.